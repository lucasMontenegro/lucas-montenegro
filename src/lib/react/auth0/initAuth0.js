import createAuth0Client from "@auth0/auth0-spa-js"
export default async function initAuth0 (setClient, setUser) {
  const client = await createAuth0Client({
    domain: process.env.REACT_APP_AUTH0_DOMAIN,
    client_id: process.env.REACT_APP_AUTH0_CLIENT_ID,
  })
  setClient(client)
  return await client.isAuthenticated() ? setUser(await client.getUser()) : setUser(null)
}