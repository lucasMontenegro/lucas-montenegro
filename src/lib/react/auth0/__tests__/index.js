import React, { useContext, useState } from "react"
import initAuth0 from "../initAuth0"
import { Auth0Context, useAuth0, Auth0Provider } from "../index.js"
import renderer from "react-test-renderer"
jest.mock(`react`, () => {
  const React = jest.requireActual("react")
  return {
    ...React,
    __esModule: true,
    default: React,
    createContext: () => ({
      Provider: function Auth0ContextProvider (props) {
        return <div {...props} className="Auth0ContextProvider" />
      },
    }),
    useContext: jest.fn(),
    useState: jest.fn(),
    useEffect: fn => fn(),
    useMemo: fn => fn(),
  }
})
jest.mock(`../initAuth0`, () => ({ __esModule: true, default: jest.fn() }))
jest.mock(`../makeLoginFunction`, () => ({
  __esModule: true,
  default: () => function login () {},
}))
jest.mock(`../makeLogoutFunction`, () => ({
  __esModule: true,
  default: () => function logout () {},
}))
describe(`../index.js`, () => {
  jestUtils.describeDependencies({
    deps: [
      `react`,
      `../initAuth0`,
      `../makeLoginFunction`,
      `../makeLogoutFunction`,
      `prop-types`,
    ],
    relativeBasePath: __dirname,
  })
  describe(`Auth0Context`, () => {
    it(`should be configured properly`, () => {
      expect(Auth0Context).toMatchSnapshot()
    })
  })
  describe(`useAuth0`, () => {
    it(`should run`, () => {
      const contextValue = {}
      useContext.mockReturnValueOnce(contextValue)
      expect(useAuth0()).toBe(contextValue)
      useContext.mockClear()
    })
  })
  describe.each([[{}], [null]])(`Auth0Provider (client %j)`, client => {
    let html
    beforeAll(() => {
      useState.mockReturnValueOnce([client, () => {}])
      useState.mockReturnValueOnce([{}, () => {}])
      html = renderer.create(
        <Auth0Provider onLoginPopupTimeout={() => {}} getLogoutUrl={() => {}}>
          children
        </Auth0Provider>
      )
    })
    it(`should render`, () => {
      expect(html.toJSON()).toMatchSnapshot()
    })
    it(`should initialize Auth0`, () => {
      expect(initAuth0.mock.calls).toHaveLength(1)
      initAuth0.mockClear()
    })
  })
})