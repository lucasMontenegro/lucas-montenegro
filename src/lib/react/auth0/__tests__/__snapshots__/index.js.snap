// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`../index.js Auth0Context should be configured properly 1`] = `
Object {
  "Provider": [Function],
  "displayName": "Auth0Context",
}
`;

exports[`../index.js Auth0Provider (client {}) should render 1`] = `
<div
  className="Auth0ContextProvider"
  value={
    Object {
      "login": [Function],
      "logout": [Function],
      "user": Object {},
    }
  }
>
  children
</div>
`;

exports[`../index.js Auth0Provider (client null) should render 1`] = `
<div
  className="Auth0ContextProvider"
  value={
    Object {
      "login": [Function],
      "logout": [Function],
      "user": Object {},
    }
  }
/>
`;

exports[`../index.js external dependencies prop-types should match the installed version 1`] = `"^15.7.2"`;

exports[`../index.js external dependencies react should match the installed version 1`] = `"^16.10.1"`;

exports[`../index.js local dependencies ../initAuth0 should match the saved hash 1`] = `"6c23ae1833abde752bb3cd1e9ac5146b04d7bb71e075f8a28791f1d876fb9405"`;

exports[`../index.js local dependencies ../makeLoginFunction should match the saved hash 1`] = `"a81c520f84391e459777a9f071c77f795fd14ff5b880d6979f283308a12c25a6"`;

exports[`../index.js local dependencies ../makeLogoutFunction should match the saved hash 1`] = `"6a4f80d82141900770e8a53470cb178c358e4cd24c85d26f81bebb812980f578"`;
